File: Go/reverse_integer/reverse_integer_test.go
package reverse_integer

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestReverseInteger(t *testing.T) {
	cases := []struct {
		name string
		x    int
		want int
	}{
		{name: "positive integer", x: 123, want: 321},
		{name: "negative integer", x: -123, want: -321},
		{name: "integer with trailing zeros", x: 120, want: 21},
		{name: "encounter overflow", x: -2147483648, want: 0},
		{name: "encounter overflow", x: 2147483647, want: 0},
	}

	for _, tt := range cases {
		t.Run(tt.name, func(t *testing.T) {
			actual := reverse(tt.x)
			assert.Equal(t, actual, tt.want)
		})
	}
}

File: Java/lib/src/test/java/leetcode/ReverseInteger/ReverseIntegerTest.java
package leetcode.ReverseInteger;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

class ReverseIntegerTest {
  protected Solution solution = new Solution();

  @ParameterizedTest(name = "{2}")
  @CsvSource({
    "123,321,positive integer",
    "-123,-321,negative integer",
    "120,21,integer with trailing zeros",
    "-2147483648,0,encounter overflow",
    "2147483647,0,encounter overflow",
  })
  void testReverseInteger(int num, int want, String displayName) {
    int actual = solution.reverse(num);
    assertEquals(want, actual);
  }
}
